### ISSUE TITLE
Request: Is it possible to encrypt casper-rw persistance?

### ISSUE THREAD

**Issue by onnimonni** [SevenBits/Enterprise#36]
**Short Description of Problem:**

I would want to have encrypted persistent file system on the usb. I might accidentally save a password in browser while using linux from the usb and I don't want anyone accessing that in case I lose my usb drive. Would it be possible to add option to use LUKS (or other encryption) in enterprise boot options?

**Comment by SevenBits** [comment]
@onnimonni Yes, I'm sure it's possible, but I've never done this myself, so I don't know the procedure. I can't imagine it would be _too_ difficult, though.

**Comment by Morg42** [comment]
I tried doing this, but didn't succeed.

What I managed to achieve was booting with partition-based encrypted persistence (which works for me as well).

This needs 2 things:
- encryption setup. I resized the FAT-partition with gparted to leave the desired space on the drive unused (2GB). Then I continued more or less along the kali howto for persistent USB overlays (kali.org/prst).  Result: a USB drive which still has Mac-/Win-accessible FAT-partition (with all remaining disk space) and a "hidden" partition for encrypted storage. (to be honest- didn't try reading it on windows. not sure if windows handles multiple partitions on a USB stick).

- tweaking startup options. If you manually add "persistent=cryptsetup persistence-encryption=luks persistence" as custom startup option, it loads regularly and asks for unlock password just after kernel loading.

You can add these options in a (third?) boot entry in enterprise.cfg (which I did, so I keep 1-live, 2-persistent, 3-install).

Maybe it would be possible to add another option to the enterprise loader (persistence / encrypted persistence), so one would not have to hassle with manual input (real fun if backspace doesn't work... :D ) or fiddling with enterprise.cfg
